src/app/leaves/actions/actions.ts

```diff
@@ -0,0 +1,27 @@
+"use server";
+import { z } from "zod";
+import { getSupabaseClient } from "../../../../utils/supabase/supabaseClient";
+import { leaveRequestSchema } from "../add/components/LeaveRequestFormBuilder";
+import { redirect } from "next/navigation";
+import { LeaveRequestStatus } from "@/constant/constant";
+
+export const onCreateLeaveRequest = async (
+  values: z.infer<typeof leaveRequestSchema>
+) => {
+
+  const supabase = await getSupabaseClient();
+  const { error } = await supabase.from("leaves").insert({
```

Comment:
Can we use enums for tables here?

Reply:
Yes, I have fixed this by creating an enum.